src/util/platform.rs

```diff
@@ -0,0 +1,648 @@
+
+    // NOTE: Using stdlib Command directly is necessary for interactive REPL
+    // Interactivity implodes otherwise.
+    use std::process::{Command as StdCommand, Stdio};
+
+    let mut command = StdCommand::new("nix");
+    command.arg("repl");
+
+    // Add installable arguments
+    for arg in installable.to_args() {
+        command.arg(arg);
+    }
+
+    // Add any extra arguments
+    for arg in extra_args {
+        command.arg(arg);
+    }
+
+    // Configure for interactive use
+    command
+        .stdin(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit());
+
+    // Execute and wait for completion
+    let status = command.status()?;
+
+    if !status.success() {
+        bail!("nix repl exited with non-zero status: {}", status);
+    }
+
+    Ok(())
+}
+
+/// Process the target specialisation based on common patterns
+pub fn process_specialisation(
+    no_specialisation: bool,
+    specialisation: Option<String>,
+    specialisation_path: &str,
+) -> Result<Option<String>> {
+    let target_specialisation =
+        handle_specialisation(specialisation_path, no_specialisation, specialisation);
+
+    debug!("target_specialisation: {target_specialisation:?}");
+
+    Ok(target_specialisation)
+}
+
+/// Execute common actions for a rebuild operation across platforms
+///
+/// This function handles the core workflow for building and managing system
+/// configurations across different platforms (`NixOS`, Darwin, Home Manager).
+/// It unifies what would otherwise be duplicated across platform-specific modules.
+///
+/// The function takes care of:
+/// 1. Properly configuring the attribute path based on platform type
+/// 2. Building the configuration
+/// 3. Handling specialisations where applicable
+/// 4. Comparing the new configuration with the current one
+///
+/// # Arguments
+///
+/// * `installable` - The Nix installable representing the configuration
+/// * `config_type` - The configuration type (e.g., "nixosConfigurations", "darwinConfigurations")
+/// * `extra_path` - Additional path elements for the attribute path
+/// * `config_name` - Optional hostname or configuration name
+/// * `out_path` - Output path for the build result
+/// * `extra_args` - Additional arguments to pass to the build command
+/// * `builder` - Optional remote builder to use
+/// * `message` - Message to display during the build process
+/// * `no_nom` - Whether to disable nix-output-monitor
+/// * `specialisation_path` - Path to read specialisations from
+/// * `no_specialisation` - Whether to ignore specialisations
+/// * `specialisation` - Optional explicit specialisation to use
+/// * `current_profile` - Path to the current system profile for comparison
+/// * `skip_compare` - Whether to skip comparing the new and current configuration
+///
+/// # Returns
+///
+/// The path to the built configuration, which can be used for activation
+#[allow(clippy::too_many_arguments)]
+pub fn handle_rebuild_workflow(
```

Comment:
I like doing a Builder for commands with many options, but this can go for now.

Reply:
What can go? The warning or the function?